sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ForeachSink.scala

```diff
@@ -30,7 +32,42 @@ import org.apache.spark.sql.{DataFrame, Encoder, ForeachWriter}
 class ForeachSink[T : Encoder](writer: ForeachWriter[T]) extends Sink with Serializable {
 
   override def addBatch(batchId: Long, data: DataFrame): Unit = {
-    data.as[T].foreachPartition { iter =>
+    // TODO: Refine this method when SPARK-16264 is resolved; see comments below.
```

Comment:
Could we just use the QueryExecution of the dataframe that is passed in, instead of creating a new one?